### PR TITLE
Fix client.cc code, since res.error() without operator overloading…

### PR DESCRIPTION
**Pull request by aleroot**
I have tried the example client.cc but wasn't compiling with latest Xcode, I had to add operator overloading to make it work.

### CODE REVIEW

**Comment by yhirose** [comment]
@aleroot, thank you for the pull request. Could you provide a unit test for the method? Then I'll start reviewing it. Thanks!

**Comment by aleroot** [comment]
I've added a simple unit test, let me know if I can help with anything else. Thanks to you for the project. This library is simple and working well.

File: httplib.h [hunk 1]
```diff
@@ -804,6 +804,12 @@ enum class Error {
   Compression,
 };
 
+std::ostream& operator << (std::ostream& os, const Error& obj)
```
Comment (yhirose) [inline httplib.h:807]: Could you put `inline` keyword at the beginning of the line?

**Review by yhirose — CHANGES_REQUESTED** [review]
Look good to me. Could you make one adjustment as below? Thanks!

**Review by yhirose — APPROVED** [review]

**Comment by yhirose** [comment]
@aleroot, thanks for your contribution!

**Pull request merged**